lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -199,6 +199,25 @@ static Value getScalarFloatValue(Value input, Location loc,
   return nullptr;
 }
 
+int64_t mlir::torch::Torch::getIntAttrAsSigned(IntegerAttr intAttr) {
+  if (intAttr.getType().isSignedInteger())
+    return intAttr.getSInt();
+  if (intAttr.getType().isUnsignedInteger())
+    return int64_t(intAttr.getUInt());
+  if (intAttr.getType().isSignlessInteger())
+    return intAttr.getInt(); // signless returns as int64_t
+  assert(false && "Unhandled integer attribute type");
+  return 0;
+}
+
+int64_t mlir::torch::Torch::getIntAttrAsIndex(IntegerAttr intAttr,
+                                              int dimSize) {
+  int64_t signedIndex = getIntAttrAsSigned(intAttr);
+  if (dimSize < 0 || signedIndex > 0)
+    return signedIndex;
+  return dimSize - (-signedIndex);
```

Comment:
Shouldn't this just be `signedIndex + dimSize`

Reply:
Yes, they're identical, of course.  I thought I was making it clearer what the intent was.  I was hoping to illustrate how negative indexes work by counting backwards from dimSize by abs(signedIndex).  If you think that's overkill, I can change to a +.  Personally, if I were looking at someone else's code, I'd find it more challenging to figure out where `signedIndex + dimSize` was coming from, especially since we don't comment at this level.

In light of this explanation, please let me know if you'd rather see the + anyway.

Reply:
Actually, never mind.  Words are always clearer.  I'll rewrite the expression your way and say what it is. :-)

Reply:
done